include/bout/snb.hxx

```diff
@@ -0,0 +1,63 @@
+#include "../options.hxx"
+#include "../invert_parderiv.hxx"
+
+/// Calculate heat flux using the Shurtz-Nicolai-Busquet (SNB) model
+///
+/// Useful references:
+///
+///   Braginskii equations by R.Fitzpatrick: http://farside.ph.utexas.edu/teaching/plasma/Plasmahtml/node35.html
+/// 
+///   J.P.Brodrick et al 2017: https://doi.org/10.1063/1.5001079 and https://arxiv.org/abs/1704.08963
+///
+///   Shurtz, Nicolai and Busquet 2000: https://doi.org/10.1063/1.1289512
+///
+class HeatFluxSNB {
+public:
+  /// Construct using the options in the "snb" section.
+  HeatFluxSNB() : HeatFluxSNB(Options::root()["snb"]) {}
+
+  /// Construct using options in given section.
+  explicit HeatFluxSNB(Options &options) {
+    invertpar = InvertPar::Create();
+
+    // Read options. Note that the defaults are initialised already
+    r = options["r"].doc("Scaling of the electron-electron mean free path")
+      .withDefault(r);
+    beta_max = options["beta_max"].doc("Maximum energy group to consider (multiple of eT)")
+      .withDefault(beta_max);
+    ngroups = options["ngroups"].doc("Number of energy groups").withDefault(ngroups);
+  }
+  
+  ~HeatFluxSNB() {
+    delete invertpar;
+  }
+
+  /// Calculate divergence of heat flux
+  /// Te: Electron temperature in eV
+  /// Ne: Electron density in m^-3
+  ///
+  /// Div_Q_SH_out : An optional output field to store the Spitzer-Harm heat flux
+  ///
+  /// Returns the divergence of heat flux in units of eV per cubic meter per second
+  /// -> multiply by e=1.602e-19 to get Watts per cubic meter.
+  Field3D divHeatFlux(const Field3D &Te, const Field3D &Ne, Field3D *Div_Q_SH_out = nullptr);
+  
+private:
+  InvertPar *invertpar; ///< Parallel inversion of tridiagonal matrices
```

Comment:
This could be `std::unique_ptr<InvertPar> invertpar{nullptr}`, with the assignment above becoming:

```cpp
  invertpar = std::unique_ptr<InvertPar>{InvertPar::Create()};
```

We can then default the destructor and move constructor and move-assignment operator:

```cpp
  HeatFluxSNB(HeatFluxSNB&&) = default;
  HeatFluxSNB& operator=(HeatFluxSNB&&) = default;
  ~HeatFluxSNB() = default;
```

We can't get away with just not mentioning them at all as we've got a default constructor. Using a `unique_ptr` also means we can't have a copy constructor, unless we do something like: https://stackoverflow.com/a/43263477/2043465